app/services/sanctu.js

```diff
@@ -134,25 +142,89 @@ export default Service.extend({
+
+      timeline.add('stagger', '-=1');
+      timeline.staggerFrom(desktopNav.find('a'), 2, { y: -10, autoAlpha: 0, ease: easing.Expo.easeOut }, 0.125, 'stagger');
+      timeline.staggerFrom(mobileNavContent.find('a'), 2, { y: -10, autoAlpha: 0, ease: easing.Expo.easeOut }, 0.125, 'stagger');
+    }
+
+    if (navStartingFromTop) {
+      mobileNav.hide();
+      let $scroll = mainContainer.find('.detectScroll');
+      $scroll.scrollTop(0);
+      mainContainer.hide();
+      let preInteractionTimeline = new TimelineLite()
+          .to(imageAnimation, 0.2, { opacity: 0, y: -10, ease: easing.Expo.easeIn })
+          .pause()
+      set(this, 'preInteractionTimeline', preInteractionTimeline);
     } else {
+      set(this, 'isPreInteraction', false);
       imageAnimation.hide();
+      let $scroll = mainContainer.find('.detectScroll');
+      $scroll.scrollTop($scroll[0].scrollHeight);
+      timeline
+        .from(desktopNav, 0.8, { y: desktopViewHeight + desktopNav.height(), ease }, "entrance")
+        .from(mainContainer, 1.2, { y: -mainContainer.outerHeight(), ease, delay: 0.2 }, "entrance")
+      timeline.play();
     }
-    if (navStartingFromTop) {
+    timeline.eventCallback("onComplete", () => this.animationCompleted());
+  },
+
+  didInteract(navShouldBeTop = true) {
+
+    const ease            = easing.Expo.easeInOut;
+    const application     = Ember.$(`.${get(this, 'applicationRouteClass')}`);
+    const mainContainer   = application.find('.liquid-container.main');
+    const timeline        = get(this, 'preInteractionTimeline');
+    let imageAnimation    = Ember.$('.GLOBAL--image-animation');
+    let desktopNav        = Ember.$('.GLOBAL--nav-bar');
+    let desktopViewHeight = Ember.$(window).height() - desktopNav.height();
+    let mobileNav         = Ember.$('.GLOBAL--mobile-nav-bar');
+    let mobileNavContent   = Ember.$('.GLOBAL--mobile-nav-content');
+
+    setTimeout(() => { set(this, 'isPreInteraction', false); }, 200);
+    if (navShouldBeTop) {
       timeline
-        .from(desktopNav, 1.2, { transform: `translateY(-${desktopNav.height()}px)` }, "entrance")
-        .from(mobileNav, 1.2, { transform: `translateY(-${mobileNav.height() + 1}px)` }, "entrance")
-        .from(mainContainer, 1.2, { transform: `translateY(${mainContainer.outerHeight()}px)` }, "entrance")
+        .to(desktopNav, 1.2, { y: 0, ease }, "entrance")
+        .fromTo(mobileNav, 1.2, { y: -mobileNav.height() }, { y: 0, ease: easing.Expo.easeOut }, "entrance")
+        .from(mainContainer, 1.2, { y: mainContainer.outerHeight(), ease, }, "entrance")
+        .to(mobileNavContent, 0, { y: mobileNav.height() + 2 }, "entrance+=0.5")
     } else {
+      let distance = mainContainer.outerHeight() - desktopNav.height();
+      let mobileDistance = mainContainer.outerHeight() - mobileNav.height();
+      // the darkest code i have ever written
+      document.OPT_OUT_OF_FIRST_BOTTOM_TO_TOP_TRANSITION = true;
```

Comment:
OPT_OUT_OF_THIS_FORMAT 😂

Reply:
https://github.com/facebook/react/blob/80bff5397bf854750dbe7c286f61654ea58938c5/src/umd/ReactUMDEntry.js#L22

Reply:
(reserved for total darkness)